lib/smart_city/dataset/business.ex

```diff
@@ -116,7 +116,9 @@ defmodule SmartCity.Dataset.Business do
     |> elem(1)
   end
 
-  defp create(map) do
-    struct(%__MODULE__{}, map)
+  defp create(%__MODULE__{} = struct) do
```

Comment:
I would do an `is_struct` guard here instead of forcing it to be the same struct? Maybe?

Reply:
That's a hard no... *Natty*.

Reply:
Not an option on our current version of elixir